### PR TITLE
plugins.atpchallenger: new plugin

### PR DESCRIPTION
**Pull request by yxiao1996**
Hi team,

This change implements one of the plugin I proposed in this issue: https://github.com/streamlink/streamlink/issues/4693. This plugin enables streamlink to scrap livestream from https://www.atptour.com/en/atp-challenger-tour/challenger-tv. 

For testing, I locally run the tests described here: https://streamlink.github.io/latest/developing.html#validating-changes
Thanks,
Yu

### CODE REVIEW

File: src/streamlink/plugins/atpchallenger.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+"""
+$description Professional tennis tournaments.
+$url www.atptour.com/en/atp-challenger-tour/challenger-tv
```
Comment (bastimeyer) [inline src/streamlink/plugins/atpchallenger.py:3]:
```suggestion
$description Tennis tournaments organized by the Association of Tennis Professionals.
$url atptour.com/en/atp-challenger-tour/challenger-tv
```

**Review by bastimeyer — CHANGES_REQUESTED** [review]
Thanks for the PR.

Streamlink already implements the livestream.com plugin (which even got rewritten recently), so the current plugin implementation is pretty much unnecessary and also wrong/incomplete, so I'm not going to comment on the code after the first HTTP request.

Due to the embedding of 3rd party streaming sites on atptour.com, the plugin needs to be turned into a wrapper plugin which just calls `return self.session.streams(url)`.

Adding new plugins which are only wrappers however is always questionable, and it depends on whether the user can get the actual streaming URL from their web browser which could then alternatively be used as Streamlink's input URL. Adding wrapper plugins for external sites overloads Streamlink with useless plugins that (currently) slow down the session initialization, so we want to avoid this. It's also problematic from the perspective of plugin maintenance.

Since there doesn't seem to be a way though to get the actual stream URL from livestream.com via the ATPtour website, I guess adding a wrapper plugin in this case is okay.

----

The following diff should be enough for a wrapper plugin and it also fixes some other issues:

```diff
diff --git a/src/streamlink/plugins/atpchallenger.py b/src/streamlink/plugins/atpchallenger.py
index 39312a9f..b0dccde8 100644
--- a/src/streamlink/plugins/atpchallenger.py
+++ b/src/streamlink/plugins/atpchallenger.py
@@ -1,6 +1,6 @@
 """
-$description Professional tennis tournaments.
-$url www.atptour.com/en/atp-challenger-tour/challenger-tv
+$description Tennis tournaments organized by the Association of Tennis Professionals.
+$url atptour.com/en/atp-challenger-tour/challenger-tv
 $type live, vod
 """
 
@@ -9,55 +9,22 @@ import re
 
 from streamlink.plugin import Plugin, pluginmatcher
 from streamlink.plugin.api import validate
-from streamlink.stream.hls import HLSStream
 
 log = logging.getLogger(__name__)
 
 
 @pluginmatcher(re.compile(
-    r"https?://(?:www\.)?atptour.com/en/atp-challenger-tour/challenger-tv"
+    r"https?://(?:www\.)?atptour\.com/(?:en|es)/atp-challenger-tour/challenger-tv"
 ))
 class AtpChallengerTour(Plugin):
-    _re_window_config = re.compile(r""".*window.config\s*=\s*(?P<json>{.*?});""", re.DOTALL)
-
     def _get_streams(self):
         iframe_url = self.session.http.get(self.url, schema=validate.Schema(
             validate.parse_html(),
-            validate.xml_xpath_string("normalize-space(.//iframe[contains(@src,'livestream.com')]/@src)")
-        ))
-        if not iframe_url:
-            return None
-
-        stream_data = self.session.http.get(iframe_url, schema=validate.Schema(
-            validate.transform(self._re_window_config.search),
-            validate.any(None, validate.all(
-                validate.get("json"),
-                validate.parse_json(),
-                {
-                    "event": {
-                        "full_name": str,
-                        "feed": {
-                            "data": list
-                        }
-                    }
-                },
-                validate.get("event")
-            ))
+            validate.xml_xpath_string(".//iframe[starts-with(@id,'vimeoPlayer_')][@src][1]/@src"),
+            validate.any(None, validate.url()),
         ))
-        if not stream_data:
-            return None
-        if len(stream_data['feed']['data']) < 1:
-            return None
-        if 'data' not in stream_data['feed']['data'][0]:
-            return None
-        if 'secure_m3u8_url' not in stream_data['feed']['data'][0]['data']:
-            return None
-        self.title = stream_data['full_name'].replace(" ", "")
-        self.category = "atp-challenger"
-        self.author = "ATP"
-        stream_url = stream_data['feed']['data'][0]['data']['secure_m3u8_url']
-
-        yield from HLSStream.parse_variant_playlist(self.session, stream_url).items()
+        if iframe_url:
+            return self.session.streams(iframe_url)
 
 
 __plugin__ = AtpChallengerTour
diff --git a/tests/plugins/test_atpchallenger.py b/tests/plugins/test_atpchallenger.py
index 2748b5de..fc35816c 100644
--- a/tests/plugins/test_atpchallenger.py
+++ b/tests/plugins/test_atpchallenger.py
@@ -6,7 +6,8 @@ class TestPluginCanHandleUrlAtpChallenger(PluginCanHandleUrl):
     __plugin__ = AtpChallengerTour
 
     should_match = [
-        'https://www.atptour.com/en/atp-challenger-tour/challenger-tv/challenger-tv-search-results/'
-        + '2022-2785-ms005-zug-alexander-ritschard-vs-dominic-stricker/2022/2785/all',
-        'https://www.atptour.com/en/atp-challenger-tour/challenger-tv'
+        "https://www.atptour.com/en/atp-challenger-tour/challenger-tv",
+        "https://www.atptour.com/es/atp-challenger-tour/challenger-tv",
+        "https://www.atptour.com/en/atp-challenger-tour/challenger-tv/challenger-tv-search-results/"
+        + "2022-2785-ms005-zug-alexander-ritschard-vs-dominic-stricker/2022/2785/all",
     ]
```

-----

Featured stream
```
$ streamlink 'https://www.atptour.com/en/atp-challenger-tour/challenger-tv'
[cli][info] Found matching plugin atpchallenger for URL https://www.atptour.com/en/atp-challenger-tour/challenger-tv
Available streams: 288p (worst), 360p (best)
```

Specific stream
```
$ streamlink 'https://www.atptour.com/en/atp-challenger-tour/challenger-tv/challenger-tv-search-results/2022-586-ms016-lexington-roman-safiullin-vs-alafia-ayeni/2022/586/all'
[cli][info] Found matching plugin atpchallenger for URL https://www.atptour.com/en/atp-challenger-tour/challenger-tv/challenger-tv-search-results/2022-586-ms016-lexington-roman-safiullin-vs-alafia-ayeni/2022/586/all
Available streams: 288p_alt2 (worst), 288p, 360p_alt2, 360p (best)
```

Unavailable stream
```
$ streamlink 'https://www.atptour.com/en/atp-challenger-tour/challenger-tv/challenger-tv-search-results/2022-586-qs018-lexington-gage-brymer-vs-divij-sharan/2022/586/all'
[cli][info] Found matching plugin atpchallenger for URL https://www.atptour.com/en/atp-challenger-tour/challenger-tv/challenger-tv-search-results/2022-586-qs018-lexington-gage-brymer-vs-divij-sharan/2022/586/all
error: Unable to open URL: https://api.new.livestream.com/accounts/5057055/events/10567785 (404 Client Error: Not Found for url: https://api.new.livestream.com/accounts/5057055/events/10567785)
```

File: tests/plugins/test_atpchallenger.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+from tests.plugins import PluginCanHandleUrl
+
+
+class TestPluginCanHandleUrlAtpChallenger(PluginCanHandleUrl):
+    __plugin__ = AtpChallengerTour
+
+    should_match = [
+        'https://www.atptour.com/en/atp-challenger-tour/challenger-tv/challenger-tv-search-results/'
+        + '2022-2785-ms005-zug-alexander-ritschard-vs-dominic-stricker/2022/2785/all',
+        'https://www.atptour.com/en/atp-challenger-tour/challenger-tv'
```
[the remark this answers is not in the document]
Comment (yxiao1996) [reply tests/plugins/test_atpchallenger.py:11]: will update

File: src/streamlink/plugins/atpchallenger.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+$type live, vod
+"""
+
+import logging
+import re
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import validate
+
+log = logging.getLogger(__name__)
```
Comment (bastimeyer) [inline src/streamlink/plugins/atpchallenger.py:13]: Please remove the logging stuff (import and `log` definition), it's not needed. Forgot to include this in my diff.

Comment (yxiao1996) [reply src/streamlink/plugins/atpchallenger.py:13]: will do

**Pull request merged**